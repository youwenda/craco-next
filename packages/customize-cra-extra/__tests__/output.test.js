const { mergeOutput, mergeOptimization, mergeWebpackResolveFallback } = require('../lib');
const { getConfig } = require('./config');
describe('merge', () => {
  test('test mergeOutput', () => {
    const developmentConfig = getConfig();
    const isEnvDevelopment = true;
    expect(
      mergeOutput({
        filename: '[name].js',
        // There are also additional JS chunk files if you use code splitting.
        chunkFilename: `chunks/[name]${isEnvDevelopment ? '' : '.[contenthash:8]'}.js`,
        assetModuleFilename: 'media/[name].[hash][ext]',
      })(developmentConfig).output
    ).toMatchSnapshot();
  });
  test('test mergeOptimization', () => {
    const developmentConfig = getConfig();
    const isEnvProduction = false;
    expect(
      mergeOptimization({
        splitChunks: {
          chunks: 'async',
          name: false,
        },
        runtimeChunk: !isEnvProduction,
      })(developmentConfig).optimization
    ).toMatchSnapshot();
  });

  test('test mergeWebpackResolveFallback', () => {
    const developmentConfig = getConfig();
    expect(
      mergeWebpackResolveFallback({
        assert: false,
        buffer: false,
        console: false,
        constants: false,
        crypto: false,
        domain: false,
        events: false,
        http: false,
        https: false,
        os: false,
        path: false,
        punycode: false,
        process: false,
        querystring: false,
        stream: false,
        string_decoder: false,
        sys: false,
        timers: false,
        tty: false,
        url: false,
        util: false,
        vm: false,
        zlib: false,
      })(developmentConfig).resolve.fallback
    ).toMatchSnapshot();
  });
});
