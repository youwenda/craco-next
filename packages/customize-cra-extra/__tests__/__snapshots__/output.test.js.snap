// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`merge test mergeOptimization 1`] = `
Object {
  "minimize": false,
  "minimizer": Array [
    TerserPlugin {
      "options": Object {
        "exclude": undefined,
        "extractComments": true,
        "include": undefined,
        "minimizer": Object {
          "implementation": [Function],
          "options": Object {
            "compress": Object {
              "comparisons": false,
              "ecma": 5,
              "inline": 2,
              "warnings": false,
            },
            "keep_classnames": false,
            "keep_fnames": false,
            "mangle": Object {
              "safari10": true,
            },
            "output": Object {
              "ascii_only": true,
              "comments": false,
              "ecma": 5,
            },
            "parse": Object {
              "ecma": 8,
            },
          },
        },
        "parallel": true,
        "test": /\\\\\\.\\[cm\\]\\?js\\(\\\\\\?\\.\\*\\)\\?\\$/i,
      },
    },
    CssMinimizerPlugin {
      "options": Object {
        "exclude": undefined,
        "include": undefined,
        "minimizer": Object {
          "implementation": [Function],
          "options": Object {},
        },
        "parallel": true,
        "test": /\\\\\\.css\\(\\\\\\?\\.\\*\\)\\?\\$/i,
        "warningsFilter": [Function],
      },
    },
  ],
  "runtimeChunk": true,
  "splitChunks": Object {
    "chunks": "async",
    "name": false,
  },
}
`;

exports[`merge test mergeOutput 1`] = `
Object {
  "assetModuleFilename": "media/[name].[hash][ext]",
  "chunkFilename": "chunks/[name].js",
  "devtoolModuleFilenameTemplate": [Function],
  "filename": "[name].js",
  "path": "craco-next/packages/customize-cra-extra/build",
  "pathinfo": true,
  "publicPath": "/",
}
`;

exports[`merge test mergeWebpackResolveFallback 1`] = `
Object {
  "assert": false,
  "buffer": false,
  "console": false,
  "constants": false,
  "crypto": false,
  "domain": false,
  "events": false,
  "http": false,
  "https": false,
  "os": false,
  "path": false,
  "process": false,
  "punycode": false,
  "querystring": false,
  "stream": false,
  "string_decoder": false,
  "sys": false,
  "timers": false,
  "tty": false,
  "url": false,
  "util": false,
  "vm": false,
  "zlib": false,
}
`;
